Mathlib/Combinatorics/CombinatorialMap.lean

```diff
@@ -0,0 +1,187 @@
+/-
+Copyright (c) 2024 Kyle Miller, Rida Hamadani. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller, Rida Hamadani
+-/
+import Mathlib.GroupTheory.Perm.Cycle.Basic
+
+/-!
+# Combinatorial Maps
+
+This module defines combinatorial maps, which are used to represent an object by its boundaries.
+
+Only two-dimensional combinatorial maps are considered.
+
+-/
+
+
+/--
+A two-dimension combinatorial map is a collection `D` of darts, and permutations `σ`, `α`, and `φ`.
+`α` is a fixed-point-free involution.
+
+Note: this definition excludes disjoint isolated vertices.
+-/
+structure CombinatorialMap (D : Type*) where
+  /--
+  Permutation `σ` whose orbits correspond to vertices. `σ` gives the next dart counter-clockwise
+  around an edge.
+  -/
+  σ : Equiv.Perm D
+  /-- Permutation `α` whose orbits correspond to edges. `α` gives the opposite dart of an edge. -/
+  α : Equiv.Perm D
+  /--
+  Permutation `φ` whose orbits correspond to faces. `φ` gives the next dart counter-clockwise of the
+  same face.
+  -/
+  φ : Equiv.Perm D
+  composition : φ * σ * α = 1
+  involutive : Function.Involutive α
```

Comment:
This should be named `edgePerm_involutive` or similar, because it's not the whole map that is involutive, just the edge permutation.